File: Python/counting_sort.py
def counting_sort(arr):
    max_val = max(arr)
    count = [0] * (max_val + 1)
    for num in arr:
        count[num] += 1
    output = []
    for i, cnt in enumerate(count):
        output.extend([i] * cnt)
    return output

File: PHP/counting_sort.php
<?php

function countingSort($array) {
    $max = max($array);
    $min = min($array);
    $range = $max - $min + 1;

    $count = array_fill(0, $range, 0);

    foreach ($array as $num) {
        $count[$num - $min]++;
    }

    $sortedArray = [];
    for ($i = 0; $i < $range; $i++) {
        while ($count[$i] > 0) {
            $sortedArray[] = $i + $min;
            $count[$i]--;
        }
    }

    return $sortedArray;
}

$array = [4, 2, 2, 8, 3, 3, 1, -8, 1, 0];
$sortedArray = countingSort($array);

echo "Sorted Array: " . implode(", ", $sortedArray) . PHP_EOL;
?>